src/algebra/tropical/big_operators.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2021 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import algebra.tropical.lattice
+import algebra.big_operators.basic
+import data.list.min_max
+
+/-!
+
+# Tropicalization of finitary operations
+
+This file provides the "big-op" or notation-based finitary operations on tropicalized types.
+This allows easy conversion between sums to Infs and prods to sums. Results here are important
+for expressing that evaluation of tropical polynomials are the minimum over a finite piecewise
+collection of linear functions.
+
+## Main declarations
+
+* `untrop_sum`
+
+## Implementation notes
+
+No concrete (semi)ring is used here, only ones with inferrable order/lattice structure, to support
+real, rat, ereal, and others (erat is not yet defined).
+
+Minima over `list α` are defined as producing a value in `with_top α` so proofs about lists do not
+directly transfer to minima over multisets or finsets.
+
+-/
+
+
+open_locale big_operators
+
+variables {R S : Type*}
+
+open tropical
+
+lemma list.trop_sum [add_monoid R] (l : list R) : trop l.sum = list.prod (l.map trop) :=
+begin
+  induction l with hd tl IH,
+  { simp },
+  { simp [←IH] }
+end
+
+lemma multiset.trop_sum [add_comm_monoid R] (s : multiset R) :
+  trop s.sum = multiset.prod (s.map trop) :=
+quotient.induction_on s (by simpa using list.trop_sum)
+
+lemma trop_sum [add_comm_monoid R] (s : finset S) (f : S → R) :
+  trop (∑ i in s, f i) = ∏ i in s, trop (f i) :=
+begin
+  cases s,
+  convert multiset.trop_sum _,
+  simp
+end
+
+lemma fintype.trop_sum [add_comm_monoid R] [fintype S] (f : S → R) :
+  trop (∑ i : S, f i) = ∏ i : S, trop (f i) :=
+trop_sum _ _
```

Comment:
This lemma seems pretty pointless to me, since it's even syntactically a match for the `trop_sum` above.

Reply:
It's shorthand for `trop_sum univ`, which is slightly more discoverable for the `i : S` instead of `i in S` syntax. I can remove if you think it is too much noise.

Reply:
I think we have more of these "specialize to univ lemmas". Looks fine to me.

Reply:
I think we only have those specializations when they remove a proof assumptions from the set version, which this one does not do.

Reply:
Aah, I see. Fair enough. Shall I cancel bors, or is it ok to have this anyway?